01.general/02.how-to-create-your-own-ad-filters/docs.en.md

```diff
@@ -2150,6 +2122,66 @@ This rule will be available for every platform except Safari extension, iOS, and
 ||example.org^
 ```
 
+
+<a id="how-to-debug"></a>
+## How to debug filtering rules
+
+It may be possible to create simple filtering rules "in your head", but for anything even slightly more complicated you'd need additional tools to debug and iterate them. There are tools to assist you with that. You can use DevTools in Chrome and its analogs in other browsers, but most AdGuard products provide another one: Filtering log.
+
+<a id="debug-filtering-log"></a>
+### Filtering log
+
+Filtering log is an advanced tool that will be helpful mostly to filter developers. It lists all web requests that pass through AdGuard, gives you exhaustive information on each of them, offers multiple sorting options, and has other useful features.
+
+Depending on which AdGuard product you're using, Filtering log can be located in different places. 
+
+* In **AdGuard for Windows** you'll find it inside *Ad Blocker* tab or via the tray menu;
+* In **AdGuard for Mac** it's under *Settings > Advanced > Filtering log*;
+* In **AdGuard for Android** it's a separate item in the side menu;
+* In **AdGuard browser extensions** it's accessible from the *Miscellaneous* settings tab or by right-clicking the extension icon.
```

Comment:
Only Chromium and Firefox based browsers filtering log show applied 
- element hiding rules(including CSS, ExtCSS)
- JS and scriptlets.